relay-event-normalization/src/normalize/mod.rs

```diff
@@ -918,6 +940,48 @@ impl Default for LightNormalizationConfig<'_> {
             span_description_rules: Default::default(),
             geoip_lookup: Default::default(),
             enable_trimming: false,
+            dynamic_measurements_config: None,
+        }
+    }
+}
+
+/// Container for global and project level [`MeasurementsConfig`]. The purpose is to handle
+/// the merging logic.
+#[derive(Clone, Debug)]
+pub struct DynamicMeasurementsConfig<'a> {
+    project: Option<&'a MeasurementsConfig>,
+    global: Option<&'a MeasurementsConfig>,
+}
+
+impl<'a> DynamicMeasurementsConfig<'a> {
+    /// Constructor for [`DynamicMeasurementsConfig`].
+    pub fn new(
+        project: Option<&'a MeasurementsConfig>,
+        global: Option<&'a MeasurementsConfig>,
+    ) -> Self {
+        DynamicMeasurementsConfig { project, global }
+    }
+
+    /// Returns an iterator over the union of [`BuiltinMeasurementKey`]s from both the global and
+    /// project-level [`MeasurementsConfig`]s.
+    pub fn builtin_measurement_keys(&'a self) -> impl Iterator<Item = &'a BuiltinMeasurementKey> {
+        let from_project = self.project.iter().flat_map(|c| &c.builtin_measurements);
+        let from_global = self.global.iter().flat_map(|c| &c.builtin_measurements);
+
+        from_project.chain(from_global).unique()
+    }
+
+    /// Gets the minimum value of the max amount of custom measurements from project and global level
+    /// [`MeasurementsConfig`] if at least one is available.
```

Comment:
Let's mention that if both exist, the most restrictive among both is taken (i.e. the minimum of the two). This may impact the outcome of light normalization.

Reply:
ok, tried to rewrite it to be more clear!